oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/config/ProviderSettings.java

```diff
@@ -229,7 +233,25 @@ public Builder oidcUserInfoEndpoint(String oidcUserInfoEndpoint) {
 		 */
 		@Override
 		public ProviderSettings build() {
-			return new ProviderSettings(getSettings());
+			Map<String, Object> settings = getSettings();
+			validateSettings(settings);
+			return new ProviderSettings(settings);
+		}
+
+		private static void validateSettings(Map<String, Object> settings) {
+			String issuer = (String) settings.get(ConfigurationSettingNames.Provider.ISSUER);
+			// rfc8414 https://datatracker.ietf.org/doc/html/rfc8414#section-2
+			// issuer is required and no queryParams and no fragment
+			Assert.hasText(issuer,"issuer is required");
+			try {
+				new URI(issuer).toURL();
+			} catch (Exception ex) {
+				throw new IllegalArgumentException("issuer must be a valid URL", ex);
+			}
+			UriComponents issuerUri = UriComponentsBuilder.fromUriString(issuer).build();
```

Comment:
This validation should be in `OAuth2AuthorizationServerConfigurer.validateProviderSettings()`

Reply:
As your opinion, I will improve it, thanks again.